lighthouse-core/lib/network-request.js

```diff
@@ -0,0 +1,244 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Fills most of the role of NetworkManager and NetworkRequest classes from DevTools.
+ * @see https://cs.chromium.org/chromium/src/third_party/blink/renderer/devtools/front_end/sdk/NetworkRequest.js
+ * @see https://cs.chromium.org/chromium/src/third_party/blink/renderer/devtools/front_end/sdk/NetworkManager.js
+ */
+
+const URL = require('./url-shim');
+const resourceTypes = require('../../third-party/devtools/ResourceType').TYPES;
+
+module.exports = class NetworkRequest {
+  constructor() {
+    this.requestId = '';
+    this._requestId = '';
```

Comment:
as mentioned, i'd love if a followup normalized these paths.

Reply:
yeah for sure 👍